File: backend/src/main/java/com/team05/petmeeting/domain/animal/dto/AnimalRes.java
package com.team05.petmeeting.domain.animal.dto;

import com.team05.petmeeting.domain.animal.entity.Animal;

public record AnimalRes(
        Long animalId,
        String desertionNo,
        String processState,
        String noticeNo,
        String noticeEdt,

        String upKindNm,
        String kindFullNm,
        String colorCd,
        String age,
        String weight,
        String sexCd,

        String popfile1,
        String popfile2,

        String careNm,
        String careTel,
        String careAddr,

        Integer totalCheerCount,
        double temperature,

        String care_reg_no
) {
    public AnimalRes(Animal animal) {
        this(
                animal.getId(),
                animal.getDesertionNo(),
                animal.getProcessState(),
                animal.getNoticeNo(),
                String.valueOf(animal.getNoticeEdt()),

                animal.getUpKindNm(),
                animal.getKindFullNm(),
                animal.getColorCd(),
                animal.getAge(),
                animal.getWeight(),
                animal.getSexCd(),

                animal.getPopfile1(),
                animal.getPopfile2(),

                animal.getCareNm(),
                animal.getCareTel(),
                animal.getCareAddr(),

                animal.getTotalCheerCount(),
                animal.getTotalCheerCount() / (double) 50, // 목표하트수 50 임시설정값

                animal.getShelter().getCareRegNo()
        );
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/animal/dto/AnimalRes.kt
package com.team05.petmeeting.domain.animal.dto

import com.team05.petmeeting.domain.animal.entity.Animal

@JvmRecord
data class AnimalRes(
    val animalId: Long,
    val desertionNo: String,
    val processState: String?,
    val noticeNo: String?,
    val noticeEdt: String,

    val upKindNm: String,
    val kindFullNm: String,
    val colorCd: String?,
    val age: String?,
    val weight: String?,
    val sexCd: String?,

    val popfile1: String,
    val popfile2: String?,

    val careNm: String?,
    val careTel: String?,
    val careAddr: String?,

    val totalCheerCount: Int,
    val temperature: Double,

    val care_reg_no: String?
) {
    constructor(animal: Animal) : this(
        animal.id,
        animal.desertionNo,
        animal.processState,
        animal.noticeNo,
        animal.noticeEdt.toString(),

        animal.upKindNm,
        animal.kindFullNm,
        animal.colorCd,
        animal.age,
        animal.weight,
        animal.sexCd,

        animal.popfile1,
        animal.popfile2,

        animal.careNm,
        animal.careTel,
        animal.careAddr,

        animal.totalCheerCount,
        animal.getTemperature(),

        animal.shelter?.careRegNo
    )
}
